src/gsy_e/models/strategy/future/strategy.py

```diff
@@ -61,6 +61,14 @@ def update(self, market: "FutureMarkets", strategy: "BaseStrategy") -> None:
                 if strategy.are_bids_posted(market.id, time_slot):
                     strategy.update_bid_rates(market, self.get_updated_rate(time_slot), time_slot)
 
+    def delete_past_state_values(self, current_market_time_slot: DateTime) -> None:
+        """Delete irrelevant values from buffers for unneeded markets."""
+        self.initial_rate.pop(current_market_time_slot, None)
+        self.final_rate.pop(current_market_time_slot, None)
+        self.energy_rate_change_per_update.pop(current_market_time_slot, None)
+        self.update_counter.pop(current_market_time_slot, None)
+        self.market_slot_added_time_mapping.pop(current_market_time_slot, None)
```

Comment:
Sorry for nitpicking a bit here, but in order to avoid code duplication one more method could be created, called `delete_market_slot_data` , better located in `TemplateStrategyUpdaterBase`, that will include all these `pop()` calls and will be responsible for popping one market slot from all dicts. That way you will be able to reuse it in `FutureTemplateStrategyOfferUpdater.delete_past_state_values`, `FutureTemplateStrategyBidUpdater.delete_past_state_values`, but also in `TemplateStrategyUpdaterBase.delete_past_state_values` (which also could benefit from this). Thanks for bearing with me, I think that this will improve the code reuse a lot!

Reply:
Thanks for the proposal! it should now be addressed and looking much better.